Crudspec (#6)
graphqlcrud/spec: docs/getting-started.md

> GraphQL CRUD is an superset of the GraphQL giving developers

Reviewer: ```suggestion
GraphQL CRUD is a superset of the GraphQL giving developers
```

> patterns for building standard data driven GraphLQL APIs.

Reviewer: ```suggestion
patterns for building standard data driven GraphQL APIs.
```

GraphQLCRUD is a fully [GraphQL compliant](http://facebook.github.io/graphql/) query language that provides patterns for accessing and modification of the data. 
For example, this GraphQLCRUD query retrieves a single user:

```graphql
{
  user(where: { id: 4 }) {
    name
  }
}
```

returns:

```json
{
  "user": {
    "name": "Mark Zuckerberg"
  }
}
```

## Rationale

GraphQL is a flexible query language supporting many different data access patterns. 

> For most of the projects CRUD operations turn out to be a very common pattern. Definifing this very common pattern enables the community to build tooling specific to the common CRUD style API.

Reviewer: ```suggestion
For most of the projects CRUD operations turn out to be a very common pattern. Defining this very common pattern enables the community to build tooling specific to the common CRUD style API.
```

> ## Targets of this spec

Reviewer: Mixing terminology - is it a spec or is it not?

1. Define minimal subset of the CRUD capabilities 
that every application or developer can consume.
2. Provide overview of the data access methods that are used to fetch data
3. Avoid corner cases or specifics of individual CRUD implementations 
4. Define lenient standard based on practices, existing API and providers
that do not enforce specific query or mutation names etc.
5. Provide set of the reference implementations for *different programming languages* in case you want to adopt reference
implementation

> 6. Provide capabilities available in vanila GraphQL (no preprocessors, helpers, annotations etc.)

Reviewer: ```suggestion
6. Provide capabilities available in vanilla GraphQL (no preprocessors, helpers, annotations etc.)
```

## Non targets of this spec

1. Define every possible method to map database or SQL Capabilities

Over the years we seen issues with developers adopting very open CRUD capabilities on the client. 

> There is no silverbullet that will give developers both flexiblity of the query capabilities on the client 

Reviewer: ```suggestion
There is no silver bullet that will give developers both flexiblity of the query capabilities on the client 
```

> For general rules for GraphQL schemas please ses: https://graphql-rules.com 

Reviewer: ```suggestion
For general rules for GraphQL schemas see [GraphQL Rules](https://graphql-rules.com)
```
